src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -629,18 +630,41 @@ private boolean isJava8() {
     ClassName tacoBellTaco = ClassName.get("com.taco.bell", "Taco");
     ClassName fishTaco = ClassName.get("org.fish.taco", "Taco");
     TypeSpec typeSpec = TypeSpec.classBuilder("Taco")
-            .superclass(fishTaco)
-            .addSuperinterface(ParameterizedTypeName.get(ClassName.get(Comparable.class), javapoetTaco))
-            .addSuperinterface(tacoBellTaco)
-            .build();
+        .superclass(fishTaco)
+        .addSuperinterface(ParameterizedTypeName.get(ClassName.get(Comparable.class), javapoetTaco))
+        .addSuperinterface(tacoBellTaco)
+        .build();
+    assertThat(toString(typeSpec)).isEqualTo(""
+        + "package com.squareup.tacos;\n"
+        + "\n"
+        + "import java.lang.Comparable;\n"
+        + "\n"
+        + "class Taco extends org.fish.taco.Taco "
+        + "implements Comparable<Taco>, com.taco.bell.Taco {\n"
+        + "}\n");
+  }
+
+  @Test public void classImplementsNestedClass() throws Exception {
+    ClassName outer = ClassName.get(tacosPackage, "Outer");
+    ClassName inner = outer.nestedClass("Inner");
+    ClassName callable = ClassName.get(Callable.class);
+    TypeSpec typeSpec = TypeSpec.classBuilder("Outer")
+        .superclass(ParameterizedTypeName.get(callable,
+            inner))
+        .addType(TypeSpec.classBuilder("Inner")
+            .addModifiers(Modifier.STATIC)
+            .build())
+        .build();
+
     assertThat(toString(typeSpec)).isEqualTo(""
-            + "package com.squareup.tacos;\n"
-            + "\n"
-            + "import java.lang.Comparable;\n"
-            + "\n"
-            + "class Taco extends org.fish.taco.Taco "
-            + "implements Comparable<Taco>, com.taco.bell.Taco {\n"
-            + "}\n");
+        + "package com.squareup.tacos;\n"
+        + "\n"
+        + "import java.util.concurrent.Callable;\n"
+        + "\n"
+        + "class Outer extends Callable<Outer.Inner> {\n"
```

Comment:
This fixes a regression. Without this PR this incorrectly emits `extends Callable<Inner>` which doesn’t work because `Inner` isn’t in scope yet.